tests/catch/unit/memory/hipHostRegister.cc

```diff
@@ -156,5 +152,56 @@ TEMPLATE_TEST_CASE("Unit_hipHostRegister_Memcpy", "",
 
   free(A);
   free(Bh);
-  hipFree(Bd);
+  HIP_CHECK(hipFree(Bd));
+}
+
+template <typename T> __global__ void fill_kernel(T* dataPtr, T value) {
+  size_t tid{blockIdx.x * blockDim.x + threadIdx.x};
+  dataPtr[tid] = value;
+}
+
+TEMPLATE_TEST_CASE("Unit_hipHostRegister_Negative", "", int, float, double) {
+  TestType* hostPtr = nullptr;
+
+  size_t sizeBytes = 1 * sizeof(TestType);
+  SECTION("hipHostRegister Negative Test - nullptr") {
+    HIP_CHECK_ERROR(hipHostRegister(hostPtr, 1, 0), hipErrorInvalidValue);
+  }
+
+// Different error codes are reported for AMD and Nvidia devices
+#if HT_AMD
+  auto errorCode = hipErrorOutOfMemory;
```

Comment:
I think we have a fix for this coming soon.

Reply:
The test is still failing as of today with the most recent changes from hipamd. I will check again next week.

Reply:
I updated the PR to change all the tests that have been fixed. Unfortunately, I'm unable to test the change because the most recent hipamd build is not building.